securesystemslib/serialization.py

```diff
@@ -0,0 +1,250 @@
+"""Serialization module provides abstract base classes and concrete
+implementations to serialize and deserialize objects.
+"""
+
+import abc
+import json
+import tempfile
+from typing import Any, Dict, Optional
+
+from securesystemslib.exceptions import DeserializationError, SerializationError
+from securesystemslib.storage import FilesystemBackend, StorageBackendInterface
+from securesystemslib.util import persist_temp_file
+
+
+# TODO: Use typing.Protocol post python 3.7
+# See https://github.com/in-toto/securesystemslib/issues/10.
```

Comment:
FYI: I just closed this issue. Let's just stick to abc for the time being and remove the comment.

Reply:
Ping

Reply:
Ping